keras/src/layers/preprocessing/image_preprocessing/equalization.py

```diff
@@ -0,0 +1,199 @@
+from keras.src import backend
+from keras.src.api_export import keras_export
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    BaseImagePreprocessingLayer,
+)
+
+
+@keras_export("keras.layers.Equalization")
+class Equalization(BaseImagePreprocessingLayer):
+    """Preprocessing layer for histogram equalization on image channels.
+
+    Histogram equalization is a technique to adjust image intensities to
+    enhance contrast by effectively spreading out the most frequent
+    intensity values. This layer applies equalization on a channel-wise
+    basis, which can improve the visibility of details in images.
+
+    This layer works with both grayscale and color images, performing
+    equalization independently on each color channel. At inference time,
+    the equalization is consistently applied.
+
+    **Note:** This layer is safe to use inside a `tf.data` pipeline
+    (independently of which backend you're using).
+
+    Args:
+        value_range: Optional list/tuple of 2 floats specifying the lower
+            and upper limits of the input data values. Defaults to `[0, 255]`.
+            If the input image has been scaled, use the appropriate range
+            (e.g., `[0.0, 1.0]`). The equalization will be scaled to this
+            range, and output values will be clipped accordingly.
+        bins: Integer specifying the number of histogram bins to use for
+            equalization. Defaults to 256, which is suitable for 8-bit images.
+            Larger values can provide more granular intensity redistribution.
+
+    Inputs: 3D (HWC) or 4D (NHWC) tensor, with float or int dtype. Input pixel
```

Comment:
We should support `channels_first` as well (since it is generally the preferred format for PyTorch). Please add `data_format=None` argument to the constructor

Reply:
Done!